compiler/testData/diagnostics/tests/delegatedProperty/recursiveType.fir.kt

```diff
@@ -4,12 +4,12 @@
 
 import kotlin.reflect.KProperty
 
-val a by <!UNRESOLVED_REFERENCE!>a<!>
+val a by <!RECURSION_IN_IMPLICIT_TYPES!>a<!>
 
 val b by Delegate(b)
 
-val c by <!UNRESOLVED_REFERENCE!>d<!>
-val d by <!UNRESOLVED_REFERENCE!>c<!>
+val c by d
```

Comment:
The problems with delegate is not reported here because `d` is not initialized. But unitialized variable is not reported as for now. If we want to avoid such regression for the time being. I can make the delegate checker report (redundant) issues with delegate. That would introduce more redundant reporting elsewhere though.